server.go

```diff
@@ -607,6 +637,23 @@ func (s *Server) removeConn(c io.Closer) {
 	}
 }
 
+func (s *Server) getCodec(contentSubtype string) (Codec, error) {
+	// TODO: do we really want to do string operations each time?
+	// This may be an extra allocation
+	// See comment in CustomCodecs
+	contentSubtype = strings.TrimSpace(strings.ToLower(contentSubtype))
```

Comment:
you might also want to push this into `stream.ContentSubtype()` and memoize so you don't do this multiple times per call.